Adding Mailhog as a transport, for local development.

tineke-corin opened the pull request.

tineke-corin committed:
Adding Mailhog as a transport, for local development.

markstuart commented on lib/transports/mailhog.js:
```diff
@@ -0,0 +1,23 @@
+    const transporter = nodemailer.createTransport({
+        host: 'localhost',
+        port: 1025,
+    });
```
Allowing these to come in from the options might be nice:
```js
const defaults = {
  host: 'localhost',
  port: 1025
}
const transportOptions = Object.assign({}, defaults, options)
const transporter = nodemailer.createTransport(transportOptions)
```
Then the consumer can set their own host and port values as needed

tineke-corin replied:
Yep, nice idea - I have added this

markstuart commented on package.json:
```diff
@@ -18,14 +18,15 @@
     "loopback": "^3.22.2",
-    "@sendgrid/mail": "^6.3.1",
-    "striptags": "^2.0.4"
+    "mailhog": "^4.16.0",
```
Not sure why this would be needed as a dependency?

tineke-corin replied:
Quite right, I thought I wanted it but changed my mind 🙄

markstuart commented on README.md:
```diff
@@ -48,13 +48,34 @@ Options:
+  `docker run -p 8025:8025 -p 1025:1025 mailhog/mailhog`
+
+  You can then browse to localhost:8025 to see the emails that have been sent. If you run mailhog on non-default
+  host or port (localhost:1025) then add the host and port into the configuration in your component-config.json:
```
Nice doc update :+1:

markstuart reviewed (approved):
:shipit:

tineke-corin committed:
Remove unecessary mailhog package. If we want to write tests that
check that mailhog captures messages, we'll want it back.

Adding ability to pass non-default host and port settings for a
mailhog transport.
